src/equations/ideal_glm_mhd_multiion.jl

```diff
@@ -0,0 +1,256 @@
+# This file includes functions that are common for all AbstractIdealGlmMhdMultiIonEquations
+
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+#! format: noindent
+
+have_nonconservative_terms(::AbstractIdealGlmMhdMultiIonEquations) = True()
+
+function varnames(::typeof(cons2cons), equations::AbstractIdealGlmMhdMultiIonEquations)
+    cons = ("B1", "B2", "B3")
+    for i in eachcomponent(equations)
+        cons = (cons...,
+                tuple("rho_" * string(i), "rho_v1_" * string(i), "rho_v2_" * string(i),
+                      "rho_v3_" * string(i), "rho_e_" * string(i))...)
+    end
```

Comment:
I know that it is massive pain to change the order of the primal variable vector at this point. However, I would like to at least remark that this is somewhat inconsistent with the standard MHD equations, which place the magnetic field at the end of the primal variable vector.

Reply:
Yes, it is a different order than the one used for single-fluid MHD. The multi-component MHD equations also use a different order. In both cases, the order was chosen for convenience in the memory access and the readability of the code.

Reply:
True, I see that for multicomponent the densities are added at the end of the vector. In that case, this is a consisten generalization.